src/Tribe/Aggregator/Event.php

```diff
@@ -410,4 +411,62 @@ public static function preserve_changed_fields( $data ) {
 
 		return $data;
 	}
+
+	/**
+	 * Returns a list of events that have their unique field entry in an old format.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $origin         The origin that is currently being handled.
+	 * @param array  $old_format_ids An array of unique fields in the old format.
+	 * @param array $new_format_ids An array of unique fields in the new format.
+	 *
+	 * @return array An array of objects in the format [ <new_format_id> => { 'meta_value' => <old_format_id>, 'pos_id' => <post_id> } ]
+	 */
+	public function get_old_format_existing_ids( $origin, $old_format_ids, $new_format_ids ) {
+		global $wpdb;
+
+		$target_to_legacy_map = array_combine( $new_format_ids, $old_format_ids );
+
+		$fields = Tribe__Events__Aggregator__Record__Abstract::$unique_id_fields;
+
+		if ( empty( $fields[ $origin ] ) ) {
+			return array();
+		}
+
+		if ( empty( $old_format_ids ) ) {
+			return array();
+		}
+
+		$key       = "_{$fields[ $origin ]['target']}";
+		$post_type = Tribe__Events__Main::POSTTYPE;
+		$interval  = "('" . implode( "','", array_map( 'esc_sql', $old_format_ids ) ) . "')";
+
+		$sql = $wpdb->prepare( "
+			SELECT
+				pm.meta_value,
+				pm.post_id
+			FROM
+				{$wpdb->postmeta} pm
+			JOIN
+				{$wpdb->posts} p
+			ON
+				pm.post_id = p.ID
+			WHERE
+				p.post_type = '{$post_type}'
+				AND meta_value IN {$interval}
+				AND meta_key = %s
+		", $key );
+
+		// find events using the correct format (`target`) meta_key and the old format unique field entry
+		$results = $wpdb->get_results( $sql, OBJECT_K );
+
+		if ( empty( $results ) ) {
+			return array();
+		}
+
+		$keys = array_values( array_intersect_key( array_flip( $target_to_legacy_map ), $results ) );
```

Comment:
(2/2) ... if we also make a change here from doing `array_flip( $target_to_legacy_map )` to `array_keys( $old_format_ids )` wouldn't we get the same result? Though this line would still be pretty busy, we'd have shaved off an unneeded line of code that adds an array operation and unneeded assignment.

Reply:
The issue here is that we need [an array that will map new format IDs to the old ones](https://github.com/moderntribe/the-events-calendar/blob/f9eb36126df2257f79048a6cdec43de80728cc06/src/Tribe/Aggregator/Event.php#L424); what happens here is keeping the `meta_key` and changing the `meta_value` with the new format.
We need to track, in the array coming back from this method, how the new format unique field maps to the old format unique field.

Reply:
I don't think what I'm suggesting would change the outcome, but since the net result is the same either way let's move ahead